src/json-rpc-handlers/dwn/process-message.ts

```diff
@@ -21,21 +26,44 @@ export const handleDwnProcessMessage: JsonRpcHandler = async (
   const { dwn, dataStream } = context;
   const { target, message } = dwnRequest.params;
   const requestId = dwnRequest.id ?? uuidv4();
-
   try {
-    let reply;
+    let reply: any;
+
     const messageType =
       message?.descriptor?.interface + message?.descriptor?.method;
 
-    // When a record is deleted via `RecordsDelete`, the initial RecordsWrite is kept as a tombstone _in addition_
-    // to the RecordsDelete message. the data associated to that initial RecordsWrite is deleted. If a record was written
-    // _and_ deleted before it ever got to dwn-server, we end up in a situation where we still need to process the tombstone
-    // so that we can process the RecordsDelete.
     if (
       messageType === DwnInterfaceName.Records + DwnMethodName.Write &&
       !dataStream
     ) {
       reply = await dwn.synchronizePrunedInitialRecordsWrite(target, message);
+    } else if (
+      messageType ===
+      DwnInterfaceName.Subscriptions + DwnMethodName.Request
```

Comment:
we need to handle subscription requests differently than a normal request, because we also need to setup and manage the socket connection directly.